src/commands/help.rs

```diff
@@ -0,0 +1,90 @@
+use termion::event::{Event, Key};
+
+use crate::commands::{quit, CommandKeybind, KeyCommand};
+use crate::config::AppKeyMapping;
+use crate::context::{AppContext, QuitType};
+use crate::error::JoshutoResult;
+use crate::event::AppEvent;
+use crate::ui::widgets;
+use crate::ui::widgets::TuiHelp;
+use crate::ui::TuiBackend;
+
+pub fn help_loop(
+    context: &mut AppContext,
+    backend: &mut TuiBackend,
+    keymap_t: &AppKeyMapping,
+) -> JoshutoResult<()> {
+    context.flush_event();
+
+    let mut offset = 0;
+    let mut search_query = String::new();
+    let mut sort_by = 1;
+
+    while context.quit == QuitType::DoNot {
+        let keymap = if search_query.is_empty() {
+            widgets::get_keymap_table(keymap_t, &search_query, sort_by)
+        } else {
+            widgets::get_keymap_table(keymap_t, &search_query[1..], sort_by)
+        };
+        backend.render(TuiHelp::new(&keymap, &mut offset, &search_query));
+
+        if !context.worker_context_ref().is_busy() && !context.worker_context_ref().is_empty() {
+            context.worker_context_mut().start_next_job();
+        }
+
+        let event = match context.poll_event() {
+            Ok(event) => event,
+            Err(_) => return Ok(()),
+        };
+
+        if let AppEvent::Termion(event) = event {
```

Comment:
This should be a match statement so everything is processed.
ie. please use `process_noninteractive` for everything that is not user input.
This keeps background workers running.
example: https://github.com/kamiyaa/joshuto/blob/dev/src/util/input.rs#L49